projects/vdk-control-cli/src/vdk/internal/control/command_groups/job/create.py

```diff
@@ -63,15 +69,45 @@ def __create_cloud_job(self, team: str, name: str) -> None:
             f"Data Job with name {name} created and registered in cloud runtime by Control Service."
         )
 
-    def __create_local_job(self, team: str, name: str, job_path: str) -> None:
-        sample_job = self.__vdk_config.sample_job_directory
+    def __create_local_job(
+        self, team: str, name: str, job_path: str, sample_job_module
+    ) -> None:
+        sample_job = self.__create_sample_job_dir(sample_job_module)
         log.debug(f"Create sample job from directory: {sample_job} into {job_path}")
         cli_utils.copy_directory(sample_job, job_path)
         local_config = JobConfig(job_path)
         if not local_config.set_team_if_exists(team):
             log.warning(f"Failed to write Data Job team {team} in config.ini.")
         log.info(f"Data Job with name {name} created locally in {job_path}.")
 
+    def __create_sample_job_dir(self, sample_job_module):
+        """
+        This method generates the directory for the sample job which will be created when invoking `vdk create`.
+        Its control flow is as follows:
+         - if the configuration variable VDK_CONTROL_SAMPLE_JOB_DIRECTORY is set, it takes precedent over
+         everything else
+         - next, if a sample_job_module is passed to the method by its invoking class, it takes precedent next; the
+         purpose of this is to allow plugins and extensions to set a new default sample job, but still not overwrite
+         one which was set by the user
+         - finally, the default sample job is used
+
+        :param sample_job_module: a Python module containing the files for a sample job
```

Comment:
I think it would be better to be sample_job_path. 

sample_job_module means the user has to follow our package structure and basically insert their own code in our packages. This might lead to issues (more likely to cause conflicts, harder for them to do, what if we change our package structure). 

sample_job_path also facilitate future use-case -e.g if we support downloading sample job from http or git or creating sample job from another data job :)

Reply:
noted

Reply:
agree, path makes more sense  than file or directory(since a certain type of file), to describe location

Reply:
resolving this since it was merged in a separate PR